### PR TITLE
feat(version): Update to Elasticsearch 2.4.5

### PR DESCRIPTION
**Pull request by mheiniger**
A new Elasticsearch version, as 1.7 reached its end of life.
According to the EOL table its about time to shift to 2.4 or even 5.4:
https://www.elastic.co/support/eol

This also lifts up the `latest` version to 2.4.5.

### CODE REVIEW

File: elasticsearch.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ path:
 node:
   name: ${HOSTNAME}
 
-script.disable_dynamic: false
+script.inline: on
```
Comment (marcbachmann) [inline elasticsearch.yml]: That's still supported in ES5?

Comment (marcbachmann) [inline elasticsearch.yml]: Or only 2.*?

**Comment by marcbachmann** [comment]
I don't see any reason to stick with v2 if we can upgrade to v5.

**Comment by marcbachmann** [comment]
We should move the login to the bottom, so outside collaborators can also run the travis tests.
It belongs into a deploy step.

File: elasticsearch.yml (hunk 1 above)
Comment (mheiniger) [inline elasticsearch.yml]: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting-security.html#security-script-source
Its supported for 2 and 5 versions.

File: start.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ fi
 
 if [ "$1" = 'elasticsearch' ]; then
 	echo -e '\nStarting elasticsearch...'
-	exec elasticsearch "$@"
+	exec elasticsearch
```
Comment (marcbachmann) [inline start.sh]: Hehe, I just `cd`d into my local directory and saw that I also tried to upgrade it. In my case I remove this whole line.

**Comment by mheiniger** [comment]
https://cloud.elastic.co only allows upgrading in two steps.. from 1 to 2 and then from 2 to 5. So we have to support both of them.

**Comment by marcbachmann** [comment]
haha

**Comment by mheiniger** [comment]
Updated `.travis.yml` to detect if its a PR and to not build in this case.

**Comment by marcbachmann** [comment]
Lol. In the second last commit the after_success script failed because of a syntax error.
So the code on master never worked?

**Comment by marcbachmann** [comment]
Strange. it ran in here: https://travis-ci.org/upfrontIO/dockerfile-elasticsearch/builds/147115139

File: start.sh (hunk 2 above)
Comment (marcbachmann) [inline start.sh]: Can you remove this line completely. Then we allow to pass the arguments to the elasticsearch process.

**Comment by mheiniger** [comment]
I actually don't know why master worked sometimes.
Normally in `yml` you need to prefix a multiline block with `- |`, otherwise it only interprets the first line.

**Pull request merged**